cache/refs.go

```diff
@@ -322,30 +295,56 @@ func (cr *cacheRecord) ID() string {
 	return cr.md.ID()
 }
 
-type immutableRef struct {
-	*cacheRecord
-	triggerLastUsed bool
-	descHandlers    DescHandlers
-}
+// hold cacheRecord.mu lock before calling
+func (cr *cacheRecord) release(ctx context.Context, forceKeep bool) error {
+	if cr.refCount() > 0 {
```

Comment:
This is not a case for warning?

Reply:
No this is a normal case, it's just checking whether there are still open refs after releasing this one (which is possible because you can have multiple immutable refs open for the same cache record). The equivalent line in the previous code was here: https://github.com/moby/buildkit/blob/b2195cec759428a7775a6fe01799b961d14a81bc/cache/refs.go#L763